File: programmers/javascript/level2/숫자의표현.js
const n = 15;

const solution = (n) => {
  let answer = 0;
  for (let i = 1; i <= n; i++) {
    let sum = 0;
    for (let j = i; j <= n; j++) {
      sum += j;
      if (sum === n) {
        answer += 1;
        break;
      } else if (sum > n) {
        break;
      }
    }
  }
  return answer;
};

console.log(solution(n));

File: python/programmers/level2/숫자의 표현.py
def solution(n):
    answer = 0
    dp = [0] * 10001
    for i in range(1, n):
        dp[i] = dp[i - 1] + i

    for i in range(n - 1):
        for j in range(i + 1, n):
            if dp[j] - dp[i] == n:
                answer += 1
            elif dp[j] - dp[i] > n:
                break
    return answer + 1


print(solution(15))
